runtime/src/launch/slurm-srun/launch-slurm-srun.c

```diff
@@ -57,6 +57,23 @@ typedef enum {
   unknown
 } sbatchVersion;
 
+// Find the default tmp directory. Try getting the tmp dir from the
+// ISO/IEC 9945 env var options first, then P_tmpdir, then "/tmp"
+static const char* getTmpDir() {
+  int i;
+  const char* possibleDirsInEnv[] = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
+  for (i = 0; i < (sizeof(possibleDirsInEnv) / sizeof(char*)); i++) {
+    const char* curDir = getenv(possibleDirsInEnv[i]);
+    if (curDir != NULL) {
+      return curDir;
+    }
+  }
+#ifdef P_tmpdir
+  return P_tmpdir;
+#endif
```

Comment:
Language wise, c's fine with multiple return statements, in fact there's another in the loop above.

Whether to use multiple returns is a style choice. Using a lot of return statements can make code (especially some c code) a lot harder to read. In this case I thought it made the intent/priority clearer. 

At one point I did have: 

```C
#ifdef P_tmpdir
  return P_tmpdir;
#else
  return "/tmp";
#endif
```

but thought dropping the #else and made it clearer that "/tmp" was a last resort. 

It adds another statement that the c compiler has to parse, since it won't be gone after the preprocessor runs, but with optimizations it'll dead code eliminate the second return. 


My initial version was: 

```C
#ifdef P_tmpdir
  #define FALLBACK_TMPDIR P_tmpdir
#else
  #define FALLBACK_TMPDIR "/tmp";
#endif

// Find the default tmp directory. Try getting the tmp dir from the
// ISO/IEC 9945 env var options first, then P_tmpdir, then "/tmp"
static const char* getTmpDir() {
  int i;
  const char* possibleDirsInEnv[] = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
  for (i = 0; i < (sizeof(possibleDirsInEnv) / sizeof(char*)); i++) {
    const char* curDir = getenv(possibleDirsInEnv[i]);
    if (curDir != NULL) {
      return curDir;
    }
  }
  return FALLBACK_TMPDIR;
}
```

but I thought that made it a little harder to read since you have to swap contexts slightly to see what FALLBACK_TMPDIR is when you're reading the function